Rewrite <input> selection tests

quantizor opened the pull request:
The previous implementation was wrong and didn't follow the spec. I did a similar set of changes in the JSDOM project, but they don't seem to have made it here. (was a few months ago)

Following https://html.spec.whatwg.org/multipage/forms.html#the-input-element:dom-textarea/input-select-3

zcorpan commented on html/semantics/forms/the-input-element/input-selection.html:
```diff
@@ -0,0 +1,122 @@
+    "url",
+    "tel",
+    "password"
+  ];
```
Add the rest of the types here where the method applies but does nothing (instead of throwing). `date` etc.

yaycmyk replied:
The spec says that it show throw invalid state error though.

zcorpan replied:
Nope! I know this is confusing, I thought there was a bug in the spec at first, but I think there isn't, it's just confusing.

These are the possibilities:

* `select()` **does not apply** - `InvalidStateError` exception (e.g. `hidden`).
* `select()` **applies**, behavior is one of:
   * Select the text (if the control is a text field, e.g. `text`).
   * Do nothing (otherwise, e.g. `date`).

Look at the "bookkeeping details" for each type to see if it applies or not.

yaycmyk replied:
Fair enough, I'll make the change. Perhaps the compatibility table can be enriched a bit to show this distinction?

zcorpan replied:
Good idea. Do you want to file an issue (or do a PR) in whatwg/html?

yaycmyk replied:
@zcorpan Done: https://github.com/whatwg/html/pull/724

zcorpan replied:
Thanks! The original comment in this thread is not yet resolved, though, right?

yaycmyk replied:
No it is, the non-textual types don't support the expanded selection attributes & methods.

zcorpan replied:
Right, but `select()` still shouldn't throw for `date`, and I think that would be good to test. (You could have a separate array for these and test only `select()`.)

yaycmyk commented:
@zcorpan Okay, ready for a second look. Some extra labels were added to the PR by accident while I was rebasing.

yaycmyk commented:
@zcorpan Could you give this another look when you get a chance?

Commit:
Rewrite <input> selection tests

The previous implementation was wrong and didn't follow the spec. These
changes add granularity for select() behavior across the spread of different
input types and accurate support for selection attributes/methods such as
`selectionStart`, `setRangeText()`, etc.

yaycmyk commented:
@zcorpan Somehow my previous commit was overwritten - sorry about that. I restored the correct code, which tests for the things we discussed.

zcorpan commented:
LGTM, thank you!
